segmentation/include/pcl/segmentation/impl/unary_classifier.hpp

```diff
@@ -174,25 +174,24 @@ pcl::UnaryClassifier<PointT>::getCloudWithLabel (typename pcl::PointCloud<PointT
   // find the 'label' field index
   std::vector <pcl::PCLPointField> fields;
   int label_idx = -1;
-  pcl::PointCloud <PointT> point;
   label_idx = pcl::getFieldIndex<PointT> ("label", fields);
 
   if (label_idx != -1)
   {
-    for (std::size_t i = 0; i < in->size (); i++)
+    for (auto& point : (*in))
```

Comment:
Can this not be `const auto&`? If it is because of the cast, maybe try `reinterpret_cast<const char*>`?

Reply:
Thanks for your advice! :grinning: I'll change `auto&` to `const auto&` then.